packages/repository/src/repositories/repository.ts

```diff
@@ -37,6 +38,20 @@ export interface ExecutableRepository<T extends Model> extends Repository<T> {
   ): Promise<AnyObject>;
 }
 
+export interface TransactionalRepository<
+  T extends Entity,
+  ID,
+  Relations extends object = {}
+> extends EntityCrudRepository<T, ID, Relations> {
```

Comment:
Discussion point: Should `TransactionalRepository` extend `Repository` or `EntityCrudRepository`?

Personally, I am inclining towards `TransactionalRepository extends Repository`, because it's possible to have transactions in repositories that are backing models with no primary keys (i.e. `CrudRepository`).

@b-admike @raymondfeng thoughts?

Reply:
Yea that sounds good to me too. I am reworking it to extend from `Repository`